tools/client-gen/metadata/package_info.go

```diff
@@ -0,0 +1,71 @@
+package metadata
+
+import (
+	"fmt"
+	"unicode"
+
+	"github.com/go-softwarelab/common/pkg/to"
+)
+
+// Package gathers the details about original package and the target package for generated code.
+type Package struct {
+	Name             string
+	OriginalName     string
+	OriginalFullName string
+	IsSamePackage    bool
+	typePrinter      func(string) string
+}
+
+// NewPackage creates a new Package instance.
+func NewPackage(name string, originalName string, originalFullName string, isSamePackage bool) *Package {
+
+	pkg := &Package{
+		Name:             name,
+		OriginalName:     originalName,
+		OriginalFullName: originalFullName,
+		IsSamePackage:    isSamePackage,
+	}
+
+	typePrinter := to.IfThen(isSamePackage, pkg.printForSamePackage).ElseThen(pkg.printForExternal)
+	pkg.typePrinter = typePrinter
+
+	return pkg
+}
+
+// OriginalPkgImportStatement returns the import statement for the original package if the target package is different from the original package.
+func (pkg *Package) OriginalPkgImportStatement() string {
+	if pkg.IsSamePackage {
+		return ""
+	}
+	return fmt.Sprintf(`import "%s"`, pkg.OriginalFullName)
+}
+
+// PrintType returns the formatted type name with the package prefix if it's external or without it.
+func (pkg *Package) PrintType(typeName string) string {
+	return pkg.typePrinter(typeName)
+}
+
+func (pkg *Package) printForSamePackage(typeName string) string {
+	return typeName
+}
+
+func (pkg *Package) printForExternal(typeName string) string {
+	if typeName == "" {
+		return typeName
+	}
```

Comment:
```suggestion
	if len(typeName) == 0 {
	return ""
}
```
since later u do `typeName[0] == '*'` even this early return guarantee u will never get to `typeName[0] == '*'` checking here `len` instead of  empty string `""` it's more readable WDYT?

Reply:
From my perspective comparing to empty string is more readable and concise